internal/renderers/skia/lib.rs

```diff
@@ -60,42 +64,48 @@ pub struct SkiaRenderer {
     path_cache: ItemCache<Option<(Vector2D<f32, PhysicalPx>, skia_safe::Path)>>,
     rendering_metrics_collector: RefCell<Option<Rc<RenderingMetricsCollector>>>,
     rendering_first_time: Cell<bool>,
-    surface: DefaultSurface,
+    surface: once_cell::unsync::OnceCell<Box<dyn Surface>>,
+    window_handle_provider: Rc<dyn raw_window_handle::HasWindowHandle>,
+    display_handle_provider: Rc<dyn raw_window_handle::HasDisplayHandle>,
 }
 
 impl SkiaRenderer {
     /// Creates a new renderer is associated with the provided window adapter.
     pub fn new(
-        window_handle: raw_window_handle::WindowHandle<'_>,
-        display_handle: raw_window_handle::DisplayHandle<'_>,
-        size: PhysicalWindowSize,
+        window_handle_provider: Rc<dyn raw_window_handle::HasWindowHandle>,
+        display_handle_provider: Rc<dyn raw_window_handle::HasDisplayHandle>,
     ) -> Result<Self, PlatformError> {
-        let surface = DefaultSurface::new(window_handle, display_handle, size)?;
-
         Ok(Self {
             maybe_window_adapter: Default::default(),
             rendering_notifier: Default::default(),
             image_cache: Default::default(),
             path_cache: Default::default(),
             rendering_metrics_collector: Default::default(),
             rendering_first_time: Cell::new(true),
-            surface,
+            surface: Default::default(),
+            window_handle_provider,
+            display_handle_provider,
         })
     }
 
     /// Render the scene in the previously associated window. The size parameter must match the size of the window.
     pub fn render(&self) -> Result<(), i_slint_core::platform::PlatformError> {
+        let Some(surface) = self.surface.get() else {
+            return Err(format!("Skia Renderer: render() called before a resize event was dispatched to the slint::Window").into());
```

Comment:
don't we just ignore this error in the software renderer?

Reply:
Yes. I'd also prefer to ignore it here TBH. If you're also ok with that then I'll change it. But it is kind of a "trap" :)